usaspending_api/spending_explorer/v2/filters/type_filter.py

```diff
@@ -1,13 +1,78 @@
 from usaspending_api.financial_activities.models import FinancialAccountsByProgramActivityObjectClass
+from usaspending_api.references.models import GTASTotalObligation
 from usaspending_api.spending_explorer.v2.filters.explorer import Explorer
-from usaspending_api.common.helpers.generic_helper import generate_last_completed_fiscal_quarter
 from usaspending_api.spending_explorer.v2.filters.spending_filter import spending_filter
 
 
+UNREPORTED_DATA_NAME = 'Unreported Data*'
+VALID_UNREPORTED_DATA_TYPES = ['agency', 'budget_function', 'object_class']
+
+
+def get_unreported_data_obj(queryset, limit, spending_type, actual_total, fiscal_year, fiscal_quarter) -> \
+        (list, float):
+    """ Returns the modified list of result objects including the object corresponding to the unreported amount, only
+        if applicable. If the unreported amount does not fit within the limit of results provided, it will not be added.
+
+        Args:
+            queryset: Django queryset with all necessary filters, etc already applied
+            limit: number of results to limit to
+            spending_type: spending explorer category
+            actual_total: total calculated based on results in `queryset`
+            fiscal_year: fiscal year from request
+            fiscal_quarter: fiscal quarter from request
+
+        Returns:
+            result_set: modified (if applicable) result set as a list
+            expected_total: total calculated from GTAS
+    """
+
+    queryset = queryset[:limit] if type == 'award' else queryset
+
+    result_set = []
+    result_keys = ['id', 'code', 'type', 'name', 'amount']
+    for entry in queryset:
+        condensed_entry = {}
+        for key in result_keys:
+            condensed_entry[key] = entry[key]
+        result_set.append(condensed_entry)
+
+    expected_total = GTASTotalObligation.objects.filter(fiscal_year=fiscal_year, fiscal_quarter=fiscal_quarter). \
+        values_list('total_obligation', flat=True). \
+        first()
+
+    if spending_type in VALID_UNREPORTED_DATA_TYPES:
+        unreported_obj = {
+            'id': None,
+            'code': None,
+            'type': spending_type,
+            'name': UNREPORTED_DATA_NAME,
+            'amount': None
+        }
+
+        # if both values are actually available, then calculate the amount, otherwise leave it as the default of None
+        if not (actual_total is None or expected_total is None):
+            unreported_obj['amount'] = expected_total - actual_total
+
+            if len(result_set) == 0:
+                result_set.append(unreported_obj)
+            elif len(result_set) < limit:
+                result_set.append(unreported_obj)
+            else:
+                if unreported_obj['amount'] > result_set[-1]['amount']:
+                    result_set[-1] = unreported_obj
+
+        result_set.sort(key=operator.itemgetter('amount'), reverse=True)
```

Comment:
does a lambda expression not work here?

Reply:
I saw both in this post when I was looking for the most efficient way to sort: https://stackoverflow.com/questions/72899/how-do-i-sort-a-list-of-dictionaries-by-values-of-the-dictionary-in-python

I do like the lambda approach and can incorporate it here.